zerver/lib/emoji.py

```diff
@@ -97,9 +97,11 @@ def check_emoji_admin(user_profile: UserProfile, emoji_name: Optional[str]=None)
         raise JsonableError(_("Must be an organization administrator or emoji author"))
 
 def check_valid_emoji_name(emoji_name: str) -> None:
-    if re.match(r'^[0-9a-z.\-_]+(?<![.\-_])$', emoji_name):
-        return
-    raise JsonableError(_("Invalid characters in emoji name"))
+    if emoji_name:
+        if re.match(r'^[0-9a-z.\-_]+(?<![.\-_])$', emoji_name):
+            return
+        raise JsonableError(_("Invalid characters in emoji name"))
+    raise JsonableError(_("Please enter emoji name"))
```

Comment:
Can you change this to "Emoji name is missing"?  That's a happier error message for the API case, and works great in the UI too.

Reply:
Done